src/lib.rs

```diff
@@ -207,13 +222,14 @@ pub fn group_events<'a>(events: &'a [(usize, Event<'a>)]) -> Vec<Group<'a>> {
             // make the group self-contained.
             Event::Start(Tag::Paragraph | Tag::CodeBlock(..)) => {
                 // A translatable group starts here.
-                groups.push(state.into_group(idx, events));
+                groups.push(state.into_group(idx, events, &mut skip_next_group));
+
                 state = State::Translate(idx);
```

Comment:
I would like to avoid the `&mut bool` in-out parameter in `into_group`. I find that in-out parameters makes it hard to know what is updated where (and why it is updated).

I think we could avoid it if we instead change `state` to `State::Skip` here when `skip_next_group == true`. Perhaps we could have a pure function which takes a `State` and a Boolean and returns the new state and the new Boolean?

Reply:
Sounds good to me.  I'll add that in.  I had done it that way initially, and started getting tired of the state threading.  But given that we'll eventually expand what we're keeping track of to include translator comments and message contexts, we might as well prepare the way for it.

I've adjusted the code to manage a GroupingContext object, and have into_group() consume and produce it.